src/widgets/table/table.rs

```diff
@@ -775,7 +775,7 @@ impl Table<'_> {
             end += 1;
         }
 
-        let selected = selected.unwrap_or(0).min(self.rows.len() - 1);
+        let selected = selected.unwrap_or(start).min(self.rows.len() - 1);
```

Comment:
Not sure if I understand the unwrap here. When there is no selection, the following block gets pointless? If selection, do something, else skip ahead? Or am I missing something?

Reply:
This is an easy way to skip past the two while blocks (they both evaluate as false). In general, I never check for an empty before a for loop unless there's some behavior that needs to act differently on an empty list. This is similar.

Because this is a bug fix and not a rewrite, the simplest smallest fix that addresses the problem is the right one.

Happy for you to rewrite this if you'd like to put in the time and effort, but that's not the purpose of this PR.

Reply:
> the simplest smallest fix that addresses the problem is the right one.

The unwrap_or(start) seems more confusing to me than `if let Some(selected) = selected.map(… min …)`. It somehow magically skips the following loops? Then I need to read and understand the loops to understand what is happening. Reading an if is way simpler here: loops only handled when there is a selection.

I am not about a rewrite, I am for a simple and readable way to fix this. And I don't think the `unwrap_or(start)` is readable.

Reply:
6eb01268e